### PR TITLE
Fix: add PoolScopeResource validation and dynamic configuration support

### PR DESCRIPTION
**Pull request by LaurenceLiZhixin**
1. add PoolScopeResource validation to support multi version of pool scope resources, like v1/v1beta1 version  of endpointslices.

2. Add pool scope resource type configuration, which read from yurt-hub-cfg cm's `pool_scope_resources` key.
For example:
we can add 
```
 pool_scope_resources: '[{"Group":"","Version":"v1","Resource":"pods"}]'
```
into yurt-hub-cfg config map to let pool-coordinator cache pods resources.

3. Add etcd transaction max num to 100k.
4. Add dynamic client for pool-coordinator to support crd informer.

### CODE REVIEW

File: pkg/yurthub/poolcoordinator/resources/resources.go [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+	"k8s.io/client-go/informers"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/klog/v2"
+
+	"github.com/openyurtio/openyurt/pkg/yurthub/util"
+)
+
+type PoolScopeResourcesManger struct {
+	factory                      informers.SharedInformerFactory
```
Comment (rambohe-ch) [inline pkg/yurthub/poolcoordinator/resources/resources.go:22]: `factory` field is unused

File: pkg/yurthub/poolcoordinator/resources/resources.go [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+}
+
+func (m *PoolScopeResourcesManger) getGroupVersionVerifyFunction(client kubernetes.Interface) func(gvr schema.GroupVersionResource) (bool, string) {
+	return func(gvr schema.GroupVersionResource) (bool, string) {
+		maxRetry := 3
+		duration := time.Second * 5
+		counter := 0
+		var err error
+		for counter <= maxRetry {
+			if _, err = client.Discovery().ServerResourcesForGroupVersion(gvr.GroupVersion().String()); err == nil {
```
Comment (rambohe-ch) [inline pkg/yurthub/poolcoordinator/resources/resources.go:141]: if cloud-edge network is off, client can not verify the gvr exist or not. maybe we need to store the gvr info into local disk for yurthub restart use.

File: pkg/yurthub/poolcoordinator/resources/resources.go (hunk 1 above)
Comment (LaurenceLiZhixin) [reply pkg/yurthub/poolcoordinator/resources/resources.go:22]: fixed

File: pkg/yurthub/poolcoordinator/resources/resources.go (hunk 2 above)
Comment (LaurenceLiZhixin) [reply pkg/yurthub/poolcoordinator/resources/resources.go:141]: Yes, Shall we do that in the next release? At current release,  if cloud-edge network is off, the pool-coordinator would not work as there would be no leader. Let's do fix these issues together in the next release!
@rambohe-ch

Comment (rambohe-ch) [reply pkg/yurthub/poolcoordinator/resources/resources.go:140]: ok

**Review by rambohe-ch — APPROVED** [review]

**Pull request merged**